src/components/Navbar.tsx

```diff
@@ -0,0 +1,183 @@
+        icon={icon}
+        size="2x" // this should be a param
+        color={
+          isSelected
+            ? style.primary
+            : theme?.colorMode === "dark"
+            ? style.lightBackground
+            : style.darkBackground
+        }
+      />
+      <div className={`navbar-item-title ${isSelected ? "selected" : ""}`}>
+        <span className="navbar-item-title-text">{title}</span>
+      </div>
+    </li>
+  );
+}
+
+type NavbarProps = {
+  label: string;
+  children: React.ReactNode;
+};
+
+export function Navbar(props: NavbarProps) {
+  const [selectedId, setSelectedId] = useState<number>();
+  const [items, setItems] = useState<React.RefObject<HTMLLIElement>[]>([]);
+  const selectedIndicatorRef = useRef<HTMLDivElement>(null);
+
+  // TODO: refactor this monstrosity (maybe ask Sandrina to give some tips)
+  const animateWithCss = (
+    domMenuItem: HTMLLIElement,
+    domSelectedIndicator: HTMLDivElement
+  ) => {
+    // First time an item is selected, the icon appears
+    if (!domSelectedIndicator.style.visibility) {
+      domSelectedIndicator.style.top = `${domMenuItem.offsetHeight - 12}px`;
+      domSelectedIndicator.style.left = `${
+        // center positioning the icon
+        domMenuItem.offsetLeft +
+        domMenuItem.clientWidth / 2 -
+        domSelectedIndicator.clientWidth / 2
+      }px`;
+      domSelectedIndicator.style.visibility = "visible";
+    } else {
+      // If an item is already selected, move the icon to next position
+      const currentPosition = domSelectedIndicator.offsetLeft;
+      const nextLeft = domMenuItem.offsetLeft;
+      const movement = nextLeft - currentPosition;
+      domSelectedIndicator.style.setProperty(
+        "--translate-x-value",
+        `${
+          movement +
+          domMenuItem.clientWidth / 2 -
+          domSelectedIndicator.clientWidth / 2
+        }px`
+      );
+      domSelectedIndicator.classList.add("animate");
+    }
+  };
+
+  useLayoutEffect(() => {
+    const domMenuItem = items[selectedId!]?.current;
+    const domSelectedIndicator = selectedIndicatorRef.current;
+    if (domMenuItem && domSelectedIndicator) {
+      animateWithCss(domMenuItem, domSelectedIndicator);
+    }
+  }, [items, selectedId]);
+
+  const registerItem = useCallback((item: React.RefObject<HTMLLIElement>) => {
+    setItems((prevItems) => [...prevItems, item]);
+  }, []);
+
+  return (
+    <ThemeProvider>
+      <NavbarContext.Provider
+        value={{
+          selectedId,
+          setSelectedId,
+          registerItem,
+        }}
+      >
+        <nav aria-label={props.label}>
+          <ul role="menubar" className="navbar">
+            {React.Children.map(props.children, (child, index) => {
+              if (isNavbarItem(child)) {
+                throw Error("Only NavbarItem allowed as child of Navbar");
+              }
+              return React.cloneElement(child as React.ReactElement, {
+                id: index,
```

Comment:
bug: You might already know that we should avoid [`index` for unique ids](https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66) (or even `key` prop in arrays) has its downsides when the component re-renders. The same applies here. Ideally, the good practice here is to force the user to pass a `id` required prop for each NavItem.

Reply:
hehe yes, at least in my mind, this navbar had a fixed number of items, so I didn't consider this a problem. Also now that you mention it, I spent some time thinking about if it made sense to expose something like an id and didn't arrive at a strong conclusion

Reply:
Usually, yes a navbar has a fixed number, but think about SPAs, it's not that uncommon for the navbar to change based on user permissions, login/logout, etc etc ;)

Reply:
Ah, nevermind, i noticed now the  `defaultActive={0}`.